### PR TITLE
solar system editor: improve docs + add confirmation dlg

### LINKED ISSUE #1299: q: multiple same-named comets: what happens in this case?
I was querying MPC and received this result:

image

(see also SUG 20.0.2, fig. 12.10)

What will happen when I select these identical names? Are these different records for the same comet (but why are they presented several times?) I assume that the end result is a single, logical update; and that the same would result from selecting one of the records rather than three of them.

While we are here, I have a few more questions that have been wrestling with for a long time:

1. what is the difference between "overwrite existing objects" and "update existing objects"? what is lost when overwriting existing objects?
1. what is NOT updated when ticking "Update only the orbital elements"? 
1. the dialog says "**bold** are listed names that match..." - but in my dialog I see _no_ bold names. maybe this appears only in a pristine Stellarium version, before starting to import orbital data updates?

Please bear with me: the UI is not very clear, I just want to clarify this; I suspect that the wording could be improved (or explanative tooltips added).

thank you

### PR DESCRIPTION
**Pull request by axd1967**
### Description

- added a few tooltips
- cleaned up code
- Updated SUG
- Also added confirmation dialog to reset SSO

The MPC dialog offers radiobuttons and checkboxes that beg for questions. I've tried to solve this by rephrasing the labels.

Fixes #1299 

### Screenshots (if appropriate):

image

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] This change requires a documentation update

### How Has This Been Tested?

compile + test the branch

**Test Configuration**:
* Operating system: KUbuntu 18.04
* Graphics Card: <Manufacturer (likely Intel, NVidia, AMD?), Model (HD, Geforce, Radeon..., with model number), driver version?>

## Checklist:

- [ ] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by alex-w** [comment]
@gzotti could you review the PR?

**Comment by gzotti** [comment]
Sorry, on the weekend hopefully.

File: plugins/SolarSystemEditor/src/gui/MpcImportWindow.cpp [hunk 1]
```diff
@@ -269,39 +269,55 @@ void MpcImportWindow::addObjects()
...
 	bool overwrite = ui->radioButtonOverwrite->isChecked();
 	QList<SsoElements> approvedForUpdate;
 	for (int j = 0; j < candidatesForUpdate.count(); j++)
 	{
-		QString name = candidatesForUpdate.at(j).value("name").toString();
+		auto candidate = candidatesForUpdate.at(j);
+		QString name = candidate.value("name").toString();
 		if (checkedObjectsNames.contains(name))
 		{
+			// XXX: odd... if "overwrite" is false, data is overwritten anyway.
```
Comment (alex-w) [inline plugins/SolarSystemEditor/src/gui/MpcImportWindow.cpp:294]: Please see history of changes to investigate the correct behavior of this code chunk

File: plugins/SolarSystemEditor/src/gui/SolarSystemManagerWindow.cpp [hunk 2]
```diff
@@ -158,6 +158,16 @@ void SolarSystemManagerWindow::resetImportManual(bool show)
 	}
 }
 
+void SolarSystemManagerWindow::resetSSOdefaults()
+{
+	if (!askConfirmation())
```
Comment (alex-w) [inline plugins/SolarSystemEditor/src/gui/SolarSystemManagerWindow.cpp:163]: Please simplificate the code or unificate it

**Review by alex-w — CHANGES_REQUESTED** [review]

File: plugins/SolarSystemEditor/src/gui/MpcImportWindow.cpp (hunk 1 above)
Comment (axd1967) [reply plugins/SolarSystemEditor/src/gui/MpcImportWindow.cpp:294]: I'm looking at 178045611b16efe1d04a8bae6d3a7e46ab2f1b6d

File: plugins/SolarSystemEditor/src/SolarSystemEditor.cpp [hunk 3]
```diff
@@ -1215,6 +1218,7 @@ bool SolarSystemEditor::updateSolarSystemConfigurationFile(QList<SsoElements> ob
 				else
 				{
 					//TODO: Do what, log a message?
+					qDebug() << "Warning: cannot update magnitude for type " << type;
```
Comment (alex-w) [inline plugins/SolarSystemEditor/src/SolarSystemEditor.cpp:1221]: Why not qWarning()?

Comment (axd1967) [reply plugins/SolarSystemEditor/src/SolarSystemEditor.cpp:1221]: corrected

**Review by alex-w — APPROVED** [review]

File: guide/app_ssystem_ini.tex [hunk 4]
```diff
@@ -18,16 +18,15 @@ \section{Solar System Configuration Files}
...
+You can edit the file either manually, or with the Solar System Editor plugin (see
+Section~\ref{sec:plugins:SolarSystemEditor}). The private user file is automatically created by the Solar System Editor plugin.
 
 
 %Deprecated parameters are marked by gray background. 
 %Possible new parameters are marked by yellow background.
 %% GZ This is no longer true. Reactivate somehow? 
 
-Each object's data are described in its own section which is typically
+Each object's data is described in its own section which is typically
```
Comment (gzotti) [inline guide/app_ssystem_ini.tex:29]: data can be used with singular or plural verb. We definitely have more than one value, so the plural form "are" appears more natural to use.

File: guide/app_ssystem_ini.tex [hunk 5]
```diff
@@ -337,7 +335,7 @@ \subsubsection{Minor Planet section}
 \subsubsection{Comet section}
 \label{sec:ssystem.ini:Comet}
 
-Comets are tiny, and their outgassing and close approaches to the
+Comets are tiny, and their outgassing and close approaches to Sun and
```
Comment (gzotti) [inline guide/app_ssystem_ini.tex:338]: The orbiting around the Sun does not change the orbital elements which model a perfect 2-body system of massive sun and relatively mass-less comet. It's outgassing ("jet propulsion" of some kind), radiation pressure, and gravitational interaction with the planets which cause the changes. Please revert this one.

File: guide/plg_interfaces.tex [hunk 6]
```diff
@@ -626,48 +626,45 @@ \section{Solar System Editor Plugin}
 solar system objects (planets, their moons, minor bodies) in two files.  
 File \file{data/ssystem\_major.ini} in the installation directory contains data for the planets and their moons.
 File \file{data/ssystem\_minor.ini} contains data for minor bodies, i.e., planetoids and comets.
- The file will be taken from the user data
-directory if it also exists there, which means, users can add minor
-planets or comets as they become observable by editing this file.
+The file will be taken from the user data directory if it also exists, which means that users can add minor planets or comets as they become observable by editing this file.
```
Comment (gzotti) [inline guide/plg_interfaces.tex:629]: ... if it also exists THERE. (if the file exists, not the directory!)

File: plugins/SolarSystemEditor/src/SolarSystemEditor.hpp [hunk 7]
```diff
@@ -191,11 +196,13 @@ class SolarSystemEditor : public StelModule
...
 	bool copySolarSystemConfigurationFileTo(QString filePath);
+
 	//! Replace current minor bodies file in the user data directory.
 	//! Writes warning to logfile and returns false in case of problems.
 	bool replaceSolarSystemConfigurationFileWith(QString filePath);
+
 	//! (new 0.16)
-	//! Loads all new objects from filePath (an .ini file).
+	//! Loads all new objects from filePath (an .ini file), and updates existing objects
 	//! Default proposal is ssystem_1000comsts.ini in the installation dir.
```
Comment (gzotti) [inline plugins/SolarSystemEditor/src/SolarSystemEditor.hpp:206]: While we are at typofixing: ssystem_1000comets.ini

**Comment by gzotti** [comment]
OK. I think apart from the indicated docfixes/reversals, it seems OK.

File: guide/app_ssystem_ini.tex (hunk 5 above)
Comment (axd1967) [reply guide/app_ssystem_ini.tex:338]: Indeed, it's a bit confusing ; I've rephrased it, see upcoming commit .

As a side note, isn't this paragraph part of the solution of #1299 ?

**Comment by axd1967** [comment]
I'm still not feeling happy about https://github.com/Stellarium/stellarium/pull/1304#discussion_r514338268 because @alex-w truely points to an oddity that I was not yet able to understand (I still _assume_ there was a mixup in the names of the variables.) Such a misunderstanding might cause this PR to be bugged.

File: guide/app_ssystem_ini.tex (hunk 5 above)
Comment (gzotti) [reply guide/app_ssystem_ini.tex]: I have reviewed it here, and the paragraph is good as it was.

File: plugins/SolarSystemEditor/src/gui/MpcImportWindow.cpp (hunk 1 above)
Comment (gzotti) [reply plugins/SolarSystemEditor/src/gui/MpcImportWindow.cpp:294]: Just a short note, sorry I will not have time until next week if my full attention is required.

The "overwrite" means "replace the complete entry" (handled by ssoManager->appendToSolarSystemConfigurationFile(approvedForAddition);)

updating OTOH only touches the elements configured with the updateFlags.

**Pull request merged**